### PR TITLE
Extend profiling

### PR DESCRIPTION
**Pull request by matiasinsaurralde**
I've found that with the current implementation the main loop isn't tracked. With this tweak we could inspect `Fetch` and any other internal calls:
captura de pantalla 2018-06-07 a la s 11 13 53

We could get more useful information if we extend this, I've also added memory profiling and modified the flags in order to allow both. Previously:
```
% deno -goprof cpu -allow-net hello.ts
```
After this tweak:
```
% deno -cpuprof cpu.out -memprof mem.out -allow-net hello.ts
```

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -95,6 +94,33 @@ func Init() {
 	main_map = stringAsset("main.map")
 }
 
+func startProfiling() {
+	// var err error
```
Comment (ry) [inline main.go]: Remove

File: main.go [hunk 2]
```diff
@@ -95,6 +94,33 @@ func Init() {
 	main_map = stringAsset("main.map")
 }
 
+func startProfiling() {
```
Comment (ry) [inline main.go]: maybeStartProfiling

File: main.go [hunk 3]
```diff
@@ -95,6 +94,33 @@ func Init() {
...
+		check(err)
+	}
+}
+
+func stopProfiling() {
+	if *flagCPUProf != "" {
+		pprof.StopCPUProfile()
+	}
+	if *flagMemProf != "" {
+		err := memProfile.Close()
```
Comment (ry) [inline main.go]: nit: check(memProfile.Close())

File: main.go [hunk 4]
```diff
@@ -95,6 +94,33 @@ func Init() {
...
+		cpuProfile, err := os.Create(*flagCPUProf)
+		check(err)
+		err = pprof.StartCPUProfile(cpuProfile)
+		check(err)
+	}
+	if *flagMemProf != "" {
+		var err error
+		memProfile, err = os.Create(*flagMemProf)
+		check(err)
+		err = pprof.WriteHeapProfile(memProfile)
```
Comment (ry) [inline main.go]: nit: check(pprof.WriteHeapProfile(memProfile))

File: main.go [hunk 5]
```diff
@@ -95,6 +94,33 @@ func Init() {
 	main_map = stringAsset("main.map")
 }
 
+func startProfiling() {
+	// var err error
+	if *flagCPUProf != "" {
+		cpuProfile, err := os.Create(*flagCPUProf)
+		check(err)
+		err = pprof.StartCPUProfile(cpuProfile)
```
Comment (ry) [inline main.go]: nit: check(pprof.StartCPUProfile(cpuProfile))

**Comment by matiasinsaurralde** [comment]
@ry Updated!

**Comment by ry** [comment]
Thank you sir.

**Pull request merged**